src/player.cpp

```diff
@@ -6875,6 +6891,13 @@ std::list<item> player::use_charges( const itype_id &what, int qty,
         return res;
 
     } else if( what == "UPS" ) {
+        if( is_mounted() && mounted_creature.get()->has_flag( MF_RIDEABLE_MECH ) &&
+            mounted_creature.get()->battery_item ) {
+            auto mons = mounted_creature.get();
+            int power_drain = std::min( mons->battery_item->ammo_remaining(), qty );
+            mons->use_mech_power( -power_drain );
+            qty -= std::min( qty, power_drain );
+        }
```

Comment:
And after that `}`, the code continues to count all the other UPS charges you have. It even adds charges from UPS bionic and from advanced UPS.

Reply:
It does that anyway in the existing code? - because if there isnt enough battery charge from one source, itll use a combination of the others.
But I guess if ... if its the mech weapon firing, you cant combine power sources like that...
so I should add an early return of some sort?